src/components/form/form.ts

```diff
@@ -0,0 +1,160 @@
+ *
+ * @fires igcSubmit - Emitted when the form is submitted.
+ * @fires igcReset - Emitted when the form is reset.
+ */
+export class IgcFormComponent extends EventEmitterMixin<
+  IgcFormEventMap,
+  Constructor<LitElement>
+>(LitElement) {
+  static styles = css`
+    :host {
+      display: block;
+    }
+  `;
+
+  private _controlsWithChecked = [
+    'input',
+    'radio',
+    'igc-radio',
+    'igc-switch',
+    'igc-checkbox',
+  ];
+  private _controlsWithValue = ['input', 'textarea'];
+  private _controlsThatSubmit = ['input', 'button', 'igc-button'];
+
+  /** Specifies if the form should not to be validated on submit. */
+  @property({ type: Boolean, reflect: true }) novalidate = false;
+
+  constructor() {
+    super();
+
+    this.addEventListener('click', this.handleClick);
+  }
+
+  /** Submits the form. */
+  submit(): boolean {
+    const formData = this.getFormData();
+
+    const isValid = this.reportValidity();
+    if (!this.novalidate && !isValid) {
+      return false;
+    }
+
+    this.emitEvent('igcSubmit', { detail: formData });
+    return true;
+  }
+
+  /** Resets the form. */
+  reset() {
+    const formElements = this.getFormElements();
+    formElements.forEach((element) => {
+      const tagName = element.tagName.toLowerCase();
+      if (tagName === 'select') {
+        for (let i = 0; i < element.options.length; i++) {
+          const option = element.options[i];
+          option.selected = option.defaultSelected;
+        }
+      } else if (
+        (tagName === 'input' &&
+          (element.type === 'checkbox' || element.type == 'radio')) ||
+        (tagName !== 'input' && this._controlsWithChecked.includes(tagName))
+      ) {
+        element.checked = element.defaultChecked;
+      } else if (this._controlsWithValue.includes(tagName)) {
+        element.value = element.defaultValue;
+      }
+    });
+
+    this.emitEvent('igcReset');
+  }
+
+  private getFormElements(): any[] {
+    const slot = this.shadowRoot?.querySelector('slot');
+    const assignedElements = slot?.assignedElements({ flatten: true });
+    const formElements: any[] = [];
+    assignedElements?.forEach((element: any) => {
+      if (!element.disabled) {
+        formElements.push(element);
+      }
+      const children = Array.from(element.getElementsByTagName('*')).filter(
+        (element: any) => !element.disabled
+      );
+      formElements.push(...children);
+    });
+
+    return formElements;
+  }
+
+  /** Retrieves the data from the form in the format of a FormData object. */
+  getFormData() {
+    const formData = new FormData();
+
+    const formElements = this.getFormElements();
+    formElements.forEach((element) => {
+      const tagName = element.tagName.toLowerCase();
+      if (tagName === 'select') {
+        for (let i = 0; i < element.options.length; i++) {
+          const option = element.options[i];
+          if (option.selected) {
+            formData.append(element.name, option.text);
```

Comment:
We should use the `option.value`.
```suggestion
            formData.append(element.name, option.value);
```